Tests/L0/Gradle/_suite.ts

```diff
@@ -1315,6 +1329,62 @@ describe('gradle Suite', function () {
                 done(err);
             });
     });
-    */
 
+    class CheckstyleTestTool extends CheckstyleTool {
+        protected isEnabled() {
+            return true;
+        }
+    }
+
+    function verifyModuleResult(results: AnalysisResult[], moduleName: string , expectedViolationCount: number, expectedFileCount: number, expectedReports: string[]) {
+        var analysisResults = results.filter(ar => ar.moduleName === moduleName);
+        assert(analysisResults != null && analysisResults.length != 0 , "Null or empty array");
+        assert(analysisResults.length == 1, "The array does not have a single element");
+        var analysisResult = analysisResults[0];
+
+        assert(analysisResult.affectedFileCount === expectedFileCount, "Invalid file count");
+        assert(analysisResult.violationCount === expectedViolationCount, "Invalid violation count");
+        assert(analysisResult.resultFiles.length === expectedReports.length, "Invalid number of reports");
+
+        for (var actualReport of analysisResult.resultFiles) {
+            var reportFile = path.basename(actualReport);
+            assert(expectedReports.indexOf(reportFile) > -1, "Report not found: " + actualReport);
+        }
+    }
+
+    it('Checkstyle tool retrieves results', function (done) {
```

Comment:
I couldn't get good code coverage using L0 tests to I added some unit tests for checking that empty reports are not uploaded etc.